src/components/tasks/card/index.tsx

```diff
@@ -28,15 +32,31 @@ const Card: FC<Props> = ({
   const [assigneeProfilePic, setAssigneeProfilePic] = useState(
     `${process.env.NEXT_PUBLIC_GITHUB_IMAGE_URL}/${cardDetails.assignee}/img.png`,
   );
+  const { SUCCESS, ERROR } = ToastTypes;
   const isUserAuthorized = useContext(isUserAuthorizedContext);
+  const [taskTagLevel, setTaskTagLevel] = useState<taskItem[]>()
   const [showEditButton, setShowEditButton] = useState(false);
   const [keyLongPressed] = useKeyLongPressed();
+  const [isLoading, setIsloading] = useState<boolean>(false);
+  const ITEM_BY_ID_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/items/byitem/${cardDetails.id}`;
+  const ITEMS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/items`;
   useEffect(() => {
     const isAltKeyLongPressed = keyLongPressed === ALT_KEY;
     if (isAltKeyLongPressed) {
       setShowEditButton(true)
     }
   }, [keyLongPressed]);
+  useEffect(() => {
+    (async () => {
+      try{
+        const { requestPromise } = fetch({ url: ITEM_BY_ID_URL });
+        const { data: result } = await requestPromise
+        setTaskTagLevel(result.data)
+       } catch (err: any) {
+         toast(ERROR, err.message);
+       }
+    })()
```

Comment:
Could this function be taken outside useEffect and is it not dependent on any dependency like taskTagLevel or something?

Reply:
when component mounted, we just have to fetch some data, to fetch the initial tags with levels on the task